src/solver/algorithm/strong/puzzle.rs

```diff
@@ -106,13 +244,576 @@ where
         .context("Failed to select solution set database table.")?;
 
     Ok(db)
+
+}
+*/
+
+// THIS IS ONLY FOR TESTING PURPOSES
+struct TestDB {
+    memory: HashMap<State, BitVec<u8, Msb0>>,
 }
 
+impl TestDB {
+    fn initialize() -> Self {
+        Self {
+            memory: HashMap::new(),
+        }
+    }
+}
+
+impl KVStore for TestDB {
+    fn put<R: crate::database::Record>(&mut self, key: State, record: &R) {
+        let new = BitVec::from(record.raw()).clone();
+        self.memory.insert(key, new);
+    }
+
+    fn get(
+        &self,
+        key: State,
+    ) -> Option<&bitvec::prelude::BitSlice<u8, bitvec::prelude::Msb0>> {
+        let vec_opt = self.memory.get(&key);
+        match vec_opt {
+            None => None,
+            Some(vect) => Some(&vect[..]),
+        }
+    }
+
+    fn del(&mut self, key: State) {
+        unimplemented![];
```

Comment:
NIT: Good job using unimplemented; use parentheses and not brackets whenever a macro is not collection-oriented, though.